tests/src/comparators/test_byName.cpp

```diff
@@ -0,0 +1,55 @@
+/* test_byName.cpp
+Copyright (c) 2022 by Michael Zahniser
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+*/
+
+#include "es-test.hpp"
+
+// Include only the tested class's header.
+#include "../../source/comparators/ByName.h"
+
+// ... and any system includes needed for the test file.
+
+namespace { // test namespace
+
+// #region mock data
+class NamedObject {
+public:
+	NamedObject(int name) : name(name) {}
+	int Name() const { return name; };
+	int name;
+};
+// #endregion mock data
+
+
+
+// #region unit tests
+SCENARIO( "Test basic ByName functionality." , "[ByName]" ) {
+	NamedObject values[3] = { 10, 20, 30 };
```

Comment:
Doing a test with NamedObjects and then using numbers for them was a little confusing, but the code seems correct and the use of strings or numbers doesn't really matter for the test.

Reply:
Any changes needed? There's already a comment `// #region mock data`.